Lib/glyphsLib/builder/builders.py

```diff
@@ -134,6 +140,22 @@ def masters(self):
 
         for layer_id, glyph_name, layer_name, layer_data \
                 in supplementary_layer_data:
+            if layer_data.layerId not in master_layer_ids \
+            and layer_data.associatedMasterId not in master_layer_ids:
+                logger.warn(
+                    '{}, glyph "{}": Layer "{}" is dangling and will be '
+                    'skipped. Did you copy a glyph from a different font? If '
+                    'so, you should clean up any phantom layers not associated '
+                    'with an actual master.'.format(
+                        self.font.familyName, glyph_name, layer_data.layerId))
+                continue
+
+            if not layer_name:
```

Comment:
This condition is almost redundant with the condition on line 132 (`if layer_name is not None:`), unless if we have a layer with name == empty string.

Please remove the condition on line 132, so that all layers go into `supplementary_layer_data` and so that we get your nicely formatted warning for all dropped layers.